infping.go

```diff
@@ -4,156 +4,149 @@
+	for buff.Scan() {
+		text := buff.Text()
+		fields := strings.Fields(text)
+		// Ignore timestamp
+		if len(fields) > 1 {
+			host := fields[0]
+			data := fields[4]
+			dataSplitted := strings.FieldsFunc(data, slashSplitter)
+			// Remove ,
+			dataSplitted[2] = strings.TrimRight(dataSplitted[2], "%,")
+			sent, recv, lossp := dataSplitted[0], dataSplitted[1], dataSplitted[2]
+			min, max, avg := "", "", ""
+			// Ping times
+			if len(fields) > 5 {
+				times := fields[7]
+				td := strings.FieldsFunc(times, slashSplitter)
+				min, avg, max = td[0], td[1], td[2]
+			}
+			//log.Printf("Host:%s, loss: %s, min: %s, avg: %s, max: %s", host, lossp, min, avg, max)
+			writePoints(config, con, host, sent, recv, lossp, min, avg, max)
+		}
+	}
+	std := bufio.NewReader(stdout)
+	line, err := std.ReadString('\n')
+	perr(err)
+	log.Printf("stdout:%s", line)
 }
 
-func writePoints(config *toml.TomlTree, con *client.Client, host string, sent string, recv string, lossp string, min string, avg string, max string) {
-    db := config.Get("influxdb.db").(string)
-    loss, _ := strconv.Atoi(lossp)
-    pts := make([]client.Point, 1)
-    fields := map[string]interface{}{}
-    if min != "" && avg != "" && max != "" {
-        min, _ := strconv.ParseFloat(min, 64)
-        avg, _ := strconv.ParseFloat(avg, 64)
-        max, _ := strconv.ParseFloat(max, 64)
-        fields = map[string]interface{}{
-                "loss": loss,
-                "min": min,
-                "avg": avg,
-                "max": max,
-        }
-    } else {
-        fields = map[string]interface{}{
-                "loss": loss,
-        }
-    }
-    pts[0] = client.Point{
-        Measurement: "ping",
-        Tags: map[string]string{
-            "host": host,
-        },
-        Fields: fields,
-        Time: time.Now(),
-        Precision: "s",
-    }
-
-    bps := client.BatchPoints{
-        Points:          pts,
-        Database:        db,
-        RetentionPolicy: "default",
-    }
-    _, err := con.Write(bps)
-    if err != nil {
-        log.Fatal(err)
-    }
+func writePoints(config *toml.TomlTree, clnt client.Client, host string, sent string, recv string, lossp string, min string, avg string, max string) {
+	db := config.Get("influxdb.db").(string)
+	loss, _ := strconv.Atoi(lossp)
+	tags := map[string]string{"host": host}
+	fields := map[string]interface{}{}
+
+	if min != "" && avg != "" && max != "" {
+		min, _ := strconv.ParseFloat(min, 64)
+		avg, _ := strconv.ParseFloat(avg, 64)
+		max, _ := strconv.ParseFloat(max, 64)
+		fields = map[string]interface{}{
+			"loss": loss,
+			"min":  min,
+			"avg":  avg,
+			"max":  max,
+		}
+	} else {
+		fields = map[string]interface{}{
+			"loss": loss,
+		}
+	}
+
+	bp, _ := client.NewBatchPoints(client.BatchPointsConfig{
+		Database:  db,
+		Precision: "s",
+	})
+
+	pt, err := client.NewPoint("ping", tags, fields, time.Now())
+	if err != nil {
+		log.Fatal("Error: ", err)
+	}
+
+	bp.AddPoint(pt)
+
+	err = clnt.Write(bp)
```

Comment:
In theory the batch process should be doing X number of hosts at this point, so the write should be correct here.